### PR TITLE
redirect to rendered OWNERS docs on kubernetes.dev instead of github …

### PR DESCRIPTION
**Pull request by BenTheElder**
…source

discussed here https://kubernetes.slack.com/archives/C1TU9EB9S/p1692728318487289

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Comment by BenTheElder** [comment]
this is live after https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-k8sio-deploy-app-k8s-io/1694096087704408064

**Comment by sftim** [comment]
/retitle redirect to rendered OWNERS docs on k8s.dev instead of github source

File: apps/k8s-io/configmap-nginx.yaml [hunk 1]
```diff
@@ -269,7 +269,7 @@ data:
           rewrite ^/needs-ok-to-test$ https://github.com/pulls?q=org%3Akubernetes+org%3Akubernetes-sigs+org%3Akubernetes-csi+org%3Akubernetes-client+is%3Aopen+is%3Apr+label%3Aneeds-ok-to-test+label%3A%22cncf-cla%3A+yes%22+-label%3Aneeds-rebase redirect;
           rewrite ^/oncall$          https://storage.googleapis.com/test-infra-oncall/oncall.html redirect;
           rewrite ^/oncall-hotlist$  https://github.com/kubernetes/test-infra/search?q=label%3Akind%2Foncall-hotlist+is%3Aopen&type=Issues redirect;
-          rewrite ^/owners$          https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md redirect;
+          rewrite ^/owners$          https://www.kubernetes.dev/docs/guide/owners/ redirect;
```
Comment (BenTheElder) [inline apps/k8s-io/configmap-nginx.yaml:272]: @sftim it is kubernetes.dev. k8s.dev is a redirect to kubernetes.dev and we redirect straight to kubernetes.dev

**Pull request merged**